app/src/main/java/com/pvp/app/ui/screen/task/GeneralTaskCreationForm.kt

```diff
@@ -0,0 +1,247 @@
+            isCompleted = false,
+            scheduledAt = startDate,
+            title = title,
+            userEmail = "TODO"
+        )
+
+        _tasks.add(task)
+    }
+
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TaskForm(generalTasksManager: GeneralTasksManager) {
+    var title by remember { mutableStateOf("") }
+    var description by remember { mutableStateOf("") }
+    var startDate by remember { mutableStateOf(LocalDateTime.now()) }
+    val datePickerState = rememberDatePickerState(initialDisplayMode = DisplayMode.Input)
+    val isDatePickerDialogOpen = remember { mutableStateOf(false) }
+
+    Column(
+        modifier = Modifier.padding(16.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+        Text(
+            text = "Create a General Task",
+            style = TextStyle(
+                fontSize = 24.sp,
+                fontWeight = FontWeight.Bold,
+                letterSpacing = 0.15.sp
+            ),
+            modifier = Modifier.padding(bottom = 16.dp)
+        )
+
+        TextField(
+            value = title,
+            onValueChange = { title = it },
+            label = { Text("Title") },
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(8.dp)
+        )
+
+        TextField(
+            value = description,
+            onValueChange = { description = it },
+            label = { Text("Description") },
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(8.dp)
+        )
+
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(8.dp)
+        ) {
+            Text(
+                text = "Start Date: ${startDate.toLocalDate()}",
+                style = TextStyle(fontSize = 16.sp),
+                modifier = Modifier
+                    .weight(1f)
+                    .padding(end = 8.dp)
+            )
+
+            Button(
+                onClick = {
+                    isDatePickerDialogOpen.value = true
+                },
+                modifier = Modifier.wrapContentWidth()
+            ) {
+                Text("Pick Date")
+            }
+        }
+
+        if (isDatePickerDialogOpen.value) {
+            DatePickerDialog(
+                onDismissRequest = {
+                    isDatePickerDialogOpen.value = false
+                },
+                confirmButton = {
+                    TextButton(
+                        onClick = {
+                            isDatePickerDialogOpen.value = false
+                            val instant = datePickerState.selectedDateMillis?.let {
+                                Instant.ofEpochMilli(it)
+                            }
+                            startDate = LocalDateTime.ofInstant(instant, ZoneId.systemDefault())
```

Comment:
Separate different groups of code by a single line to increase readability.
```kotlin
isDatePickerDialogOpen.value = false // <-- group 1

val instant = datePickerState.selectedDateMillis?.let { // <-- group 2
    Instant.ofEpochMilli(it)
}

startDate = LocalDateTime.ofInstant(instant, ZoneId.systemDefault()) // <-- group 3
```
If you would transfer startDate above variable initialization group (currently group 2), then only 2 groups would exist altogether and only single separation line would be needed between variable assignation and initialization groups.
```kotlin
isDatePickerDialogOpen.value = false // <-- group 1 whole block, doing same action
startDate = LocalDateTime.ofInstant(instant, ZoneId.systemDefault())

val instant = datePickerState.selectedDateMillis?.let { // <-- group 2
    Instant.ofEpochMilli(it)
}
```